### DISCUSSION TITLE
serialport-rs Community Meeting 2023-08-17

### DISCUSSION THREAD

**Discussion by sirhcel** [serialport/serialport-rs#121] [General]
This will be the second community meeting for serialport-rs.

It is scheduled for [August 17th, 2023 at 6 PM CEST/GMT+2](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2023&month=8&day=17&hour=16&min=0&sec=0&p1=37) (Berlin time).  If you'd like to join, please notify us at https://matrix.to/#/#serialport-rs:matrix.org.

Topics for this meeting are given as comments. If you find one exciting, please up-vote it via the arrow icon.

**Comment by sirhcel** [comment]
### Read Timeouts on Windows

* @jerrywrice proposes a fix for read timeouts on windows
    * Description in [SerialPort_rs Receive_Timing_Info Test Description_V1_1.pdf](https://github.com/serialport/serialport-rs/files/12352374/SerialPort_rs.Receive_Timing_Info.Test.Description_V1_1.pdf)
    * Test utility: [receive_timing_info.rs](https://gist.github.com/sirhcel/6f52baef814a7dd7c63554356325052a)

**Reply by jerrywrice** [reply]
When I formulated the patch proposed in the referenced PDF document, I was unaware or neglected to review the earlier proposed patch source in pr #79. When this was indirectly brought to my attention today, I retested and determined that pr #79 provides a better performing patch than that which I used and documented in 'SerialPort_rs Receive_Timing_Info Test Description_V1_1.pdf' (attached to the parent comment of this Discussion item). I've revised the PDF, and uploaded the revised version with new timing numbers based on testing with pr #79's patch. Cheers.

**Comment by sirhcel** [comment]
### Issue Triage

* Let's skim through the currently open issues and PRs
* Which of them are fixes for 4.x

**Comment by jerrywrice** [comment]
Here's the revised PDF document regarding Issue #106, and which documents some testing and timing data for existing PR #79.

[SerialPort_rs Receive_Timing_Info Test Description_V1_4.pdf](https://github.com/serialport/serialport-rs/files/12354548/SerialPort_rs.Receive_Timing_Info.Test.Description_V1_4.pdf)

**Comment by jerrywrice** [comment]
If no one is actively/currently working on a feature patch for PR #79 for integration with an upcoming release version, I'm available to do this. As someone who uses serial ports for commercial software development on Windows on a daily basis, I'm of the opinion that releasing with this PR integrated will be an important step forward for this crate (at least for me!). If someone is already developing this patch, I will be happy to hear this, and curious when it will be available in an official release branch. If not and I'm tagged to work it, if there are any suggestions on the Windows feature name (that enables the PR #79 patch), maybe the suggested name(s) could be mentioned in a comment - either to PR #79 itself, or elsewhere?

While I haven't previously created a feature option patch to a cargo.toml file, I've briefly reviewed the Book documentation and it appears in a few days I should be able to formulate and test one for PR #79 (i.e. specific to Windows). Once I've tested it locally, I'll generate a new pull request with the patch (dependent of course on PR #79), and those interested can review and comment as appropriate.

**Comment by pres2300** [comment]
I just want to tip my hat in support for this update.  We have some factory machines that run Windows that run tests on devices via serial.  It could help optimize our test time without needing other workarounds.
